src/execution/reservoir_sample.cpp

```diff
@@ -283,6 +455,15 @@ void BaseReservoirSampling::SetNextEntry() {
 	num_entries_to_skip_b4_next_sample = 0;
 }
 
+void BaseReservoirSampling::ReplaceElementWithIndex(duckdb::idx_t entry_index, double with_weight) {
+
```

Comment:
add comment: This function is similar to `ReplaceElement`, except the entry index and weight *must* be defined. This function is used replace elements with specific weights in a reservoir sample that were popped/removed during a merge.